packages/flutter_tools/test/commands.shard/hermetic/attach_test.dart

```diff
@@ -58,6 +59,10 @@ final vm_service.Isolate fakeUnpausedIsolate = vm_service.Isolate(
 );
 
 void main() {
+  tearDown(() {
+    MacOSDesignedForIPadDevices.allowDiscovery = false;
+  });
```

Comment:
See `allowDiscovery` global comment.  It's similar to:
https://github.com/flutter/flutter/blob/6beafa7b16064a78a496b2b400e31920ca297c08/packages/flutter_tools/test/general.shard/tester/flutter_tester_test.dart#L43-L46